.github/workflows/ci.yml

```diff
@@ -23,12 +23,6 @@ jobs:
       - name: "Install Python"
         uses: actions/setup-python@v2
 
-      - name: "Install pre-commit"
-        run: pip install pre-commit
-
-      - name: "Run pre-commit checks"
-        run: pre-commit run --all-files
```

Comment:
We generally keep linting and static analysis jobs separate from jobs which run tests. Makes it easier to see what failed.

It looks like now pre-commit runs as part of the test jobs themselves.

Reply:
While that is a reasonable benefit it has it's downsides as well - namely that in case of failures you need to go to the CI run logs and see what exactly failed and unless you know what to search/grep for it's painful.

Reply:
There are pros and cons with everything. In general adhering to the KISS principle—simplifying logic and ensure consistency—likely out weights the cons. Note it's worth noting the errors are present in the logs, albeit maybe having to toggle an additional chevron.